src/crafting/PotionContainerChangeRecipe.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\crafting;
+
+use pocketmine\item\Item;
+use pocketmine\item\ItemFactory;
+
+class PotionContainerChangeRecipe implements BrewingRecipe{
+
+	/** @var int */
+	private $inputItemId;
+	/** @var Item */
+	private $ingredient;
+	/** @var int */
+	private $outputItemId;
+
+	public function __construct(int $inputItemId, Item $ingredient, int $outputItemId) {
+		$this->inputItemId = $inputItemId;
+		$this->ingredient = clone $ingredient;
+		$this->outputItemId = $outputItemId;
+	}
```

Comment:
This constructor can use promoted properties.

Reply:
The property can still be declared using the promoted style. You just need to keep the clone assignment in the constructor.